components/fraud_proof.md

```diff
@@ -0,0 +1,64 @@
+# Fraud Proof Algorithm for Simple Sequencer
+This is a WIP -- mostly just a few notes that I'm jotting down to not forget.
+
+## Requirements
+1. Not all blockhashes are posted up front.
+    - Blockhashes should be posted roughly every 100 L1 blocks.
+2. Only one block is evaluated during the fraud proof.
+
+## Fraud Proof Structure
+
+Fraud proofs are initialized with some constant size input. In our case this is:
+
+```python
+def init_fraud_proof(latest_blockhash: bytes, invalid_epoch_commitment: int):
```

Comment:
MUCH BETTER. Store the L1 blockhash at the same time as we store L2 blockhash proposal. `hash(BLOCKHASH, proposal)` is stored on chain.

Reply:
This means that your fraud proof program is not dependent on when you made the proposal.

Reply:
Yep that's right and great point -- that is unclear